libs/MeshKernelApi/src/MeshKernel.cpp

```diff
@@ -687,6 +687,105 @@ namespace meshkernelapi
+            }
+
+            const auto lowerLeftNUnsigned = static_cast<meshkernel::UInt>(lowerLeftN);
+            const auto lowerLeftMUnsigned = static_cast<meshkernel::UInt>(lowerLeftM);
+            const auto upperRightNUnsigned = static_cast<meshkernel::UInt>(upperRightN);
+            const auto upperRightMUnsigned = static_cast<meshkernel::UInt>(upperRightM);
+            const auto boundaryPolygon = meshKernelState[meshKernelId].m_curvilinearGrid->ComputeBoundaryToPolygon({lowerLeftNUnsigned, lowerLeftMUnsigned},
+                                                                                                                   {upperRightNUnsigned, upperRightMUnsigned});
+            ConvertPointVectorToGeometryList(boundaryPolygon, boundaryPolygons);
+        }
+        catch (...)
+        {
+            lastExitCode = HandleException();
+        }
+        return lastExitCode;
+    }
+
+    MKERNEL_API int mkernel_curvilinear_count_boundaries_as_polygons(int meshKernelId, int lowerLeftN, int lowerLeftM, int upperRightN, int upperRightM, int& numberOfPolygonNodes)
+    {
+        lastExitCode = meshkernel::ExitCode::Success;
+        try
+        {
+            if (!meshKernelState.contains(meshKernelId))
+            {
+                throw meshkernel::MeshKernelError("The selected mesh kernel id does not exist.");
+            }
+
+            if (!meshKernelState[meshKernelId].m_curvilinearGrid->IsValid())
+            {
+                throw meshkernel::MeshKernelError("Invalid curvilinear grid");
+            }
+
+            if (lowerLeftN < 0)
+            {
+                throw meshkernel::MeshKernelError("The lower left n index cannot be smaller than 0");
+            }
+            if (lowerLeftM < 0)
+            {
+                throw meshkernel::MeshKernelError("The lower left m index cannot be smaller than 0");
+            }
+            if (upperRightN < 0)
+            {
+                throw meshkernel::MeshKernelError("The upper right n index cannot be smaller than 0");
+            }
+            if (upperRightM < 0)
+            {
+                throw meshkernel::MeshKernelError("The upper right m index cannot be smaller than 0");
+            }
+
+            const auto lowerLeftNUnsigned = static_cast<meshkernel::UInt>(lowerLeftN);
+            const auto lowerLeftMUnsigned = static_cast<meshkernel::UInt>(lowerLeftM);
+            const auto upperRightNUnsigned = static_cast<meshkernel::UInt>(upperRightN);
+            const auto upperRightMUnsigned = static_cast<meshkernel::UInt>(upperRightM);
+            const auto boundaryPolygon = meshKernelState[meshKernelId].m_curvilinearGrid->ComputeBoundaryToPolygon({lowerLeftNUnsigned, lowerLeftMUnsigned},
+                                                                                                                   {upperRightNUnsigned, upperRightMUnsigned});
+            numberOfPolygonNodes = static_cast<int>(boundaryPolygon.size());
```

Comment:
We do the same computation twice, first time just for getting the buffer size? That's expensive! Can't the result be cached?

Reply:
it could, we do this for several other operations. I will write a follow up